### DISCUSSION TITLE
如何为代码块指定语言？

### DISCUSSION THREAD

**Discussion by LamGC** [mirai-mamori/Sakurairo#307] [Q&A] [answered]
编写文章的时候使用了 WP 自带的代码块功能，Iro 主题有适配代码块的语法高亮，但不知道怎么指定代码语言，现在代码块的语言在瞎识别，基本没正确过。
请问有什么方法可以为代码块指定语言的吗？

**Comment by KotoriK** [comment, marked as answer]
```html
<pre><code class="language-css">p { color: red }</code></pre>
```
参见https://highlightjs.org/usage/
https://prismjs.com/#basic-usage


原来写在Discussion里我还在issue找了好久

**Reply by LamGC** [reply]
可以了，不过跟 WP 的代码块不兼容，只能转换成 Html 了。
感谢回答。

**Reply by LamGC** [reply]
> 原来写在Discussion里我还在issue找了好久

开启了 Discussions 板块的仓库就不在 issue 里提问了嘛，把 issues 板块留给真反馈问题的。

**Reply by KotoriK** [reply]
> 可以了，不过跟 WP 的代码块不兼容，只能转换成 Html 了。
> 感谢回答。

看了下 wp古腾堡的代码块也是```<pre><code>```的结构
虽然古腾堡没有提供语言相关的区块设置，但是可以在额外CSS类里边加上

**Reply by KotoriK** [reply]
> > 原来写在Discussion里我还在issue找了好久
> 
> 开启了 Discussions 板块的仓库就不在 issue 里提问了嘛，把 issues 板块留给真反馈问题的。

合理
